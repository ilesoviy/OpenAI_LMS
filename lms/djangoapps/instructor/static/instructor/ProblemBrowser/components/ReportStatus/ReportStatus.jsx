/* global gettext */
import { Icon } from '@edx/paragon';
import classNames from 'classnames';
import * as PropTypes from 'prop-types';
import * as React from 'react';

const ReportStatus = ({
    error, succeeded, inProgress, reportPath,
}) => {
    const progressMessage = (
        <div className="msg progress">
            {gettext('Your report is being generated...')}
            <Icon hidden className={['fa', 'fa-refresh', 'fa-spin', 'fa-fw']} />
        </div>
    );

    const successMessage = (
        <div className="msg success">
            {gettext('Your report has been successfully generated.')}
            {reportPath
      && (
          <a href={reportPath}>
              <Icon hidden className={['fa', 'fa-link']} />
              {gettext('View Report')}
          </a>
      )}
        </div>
    );

    const errorMessage = (
        <div className={classNames('msg', { error })}>
            {error && `${gettext('Error')}: `}
            {error}
        </div>
    );

    return (
        <div className="report-generation-status" aria-live="polite">
            {inProgress && progressMessage}
            {error && errorMessage}
            {succeeded && successMessage}
        </div>
    );
};

ReportStatus.propTypes = {
    error: PropTypes.string,
    succeeded: PropTypes.bool.isRequired,
    inProgress: PropTypes.bool.isRequired,
    reportPath: PropTypes.string,
};

ReportStatus.defaultProps = {
    error: null,
    reportPath: null,
    reportPreview: null,
    reportName: null,
};

export default ReportStatus;
