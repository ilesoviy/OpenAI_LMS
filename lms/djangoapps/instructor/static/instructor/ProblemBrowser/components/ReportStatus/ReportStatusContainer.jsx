import { connect } from 'react-redux';
import ReportStatus from './ReportStatus';

const mapStateToProps = state => ({
    selectedBlock: state.selectedBlock,
    error: state.reportStatus.error,
    inProgress: state.reportStatus.inProgress,
    succeeded: state.reportStatus.succeeded,
    reportPath: state.reportStatus.reportPath,
    timeout: state.reportStatus.timeout,
});

export const ReportStatusContainer = connect(
    mapStateToProps,
)(ReportStatus);

export default ReportStatusContainer;
