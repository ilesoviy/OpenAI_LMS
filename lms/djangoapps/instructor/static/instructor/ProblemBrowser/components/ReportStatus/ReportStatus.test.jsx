/* global test,describe,expect */
import React from 'react';
import renderer from 'react-test-renderer';
import ReportStatus from './ReportStatus';

describe('ReportStatus component', () => {
    test('render in progress status', () => {
        const component = renderer.create(
            <ReportStatus
                error={null}
                inProgress
                reportName={null}
                reportPath={null}
                succeeded={false}
            />,
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render success status', () => {
        const component = renderer.create(
            <ReportStatus
                error={null}
                inProgress={false}
                reportName="some-report-name"
                reportPath="/some/report/path.csv"
                succeeded
            />,
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render error status', () => {
        const component = renderer.create(
            <ReportStatus
                error="some error status"
                inProgress={false}
                reportName={null}
                reportPath={null}
                succeeded={false}
            />,
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
