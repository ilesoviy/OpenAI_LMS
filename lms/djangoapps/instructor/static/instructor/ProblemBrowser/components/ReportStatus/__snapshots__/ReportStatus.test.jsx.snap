// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportStatus component render error status 1`] = `
<div
  aria-live="polite"
  className="report-generation-status"
>
  <div
    className="msg error"
  >
    Error: 
    some error status
  </div>
</div>
`;

exports[`ReportStatus component render in progress status 1`] = `
<div
  aria-live="polite"
  className="report-generation-status"
>
  <div
    className="msg progress"
  >
    Your report is being generated...
    <div>
      <span
        aria-hidden={true}
        className="fa fa-refresh fa-spin fa-fw"
        id="Icon2"
      />
    </div>
  </div>
</div>
`;

exports[`ReportStatus component render success status 1`] = `
<div
  aria-live="polite"
  className="report-generation-status"
>
  <div
    className="msg success"
  >
    Your report has been successfully generated.
    <a
      href="/some/report/path.csv"
    >
      <div>
        <span
          aria-hidden={true}
          className="fa fa-link"
          id="Icon2"
        />
      </div>
      View Report
    </a>
  </div>
</div>
`;
